elrond-wasm-node/src/api/send_api_node.rs

```diff
@@ -0,0 +1,211 @@
+use super::ArwenBigUint;
+use crate::ArwenApiImpl;
+use elrond_wasm::api::SendApi;
+use elrond_wasm::types::{Address, ArgBuffer, BoxedBytes, CodeMetadata};
+
+extern "C" {
+	fn transferValue(
+		dstOffset: *const u8,
+		valueOffset: *const u8,
+		dataOffset: *const u8,
+		dataLength: i32,
+	) -> i32;
+
+	fn transferESDT(
+		dstOffset: *const u8,
+		tokenIdOffset: *const u8,
+		tokenIdLen: i32,
+		valueOffset: *const u8,
+		gasLimit: i64,
+		dataOffset: *const u8,
+		dataLength: i32,
+	) -> i32;
+
+	fn asyncCall(dstOffset: *const u8, valueOffset: *const u8, dataOffset: *const u8, length: i32);
+
+	fn createContract(
+		gas: u64,
+		valueOffset: *const u8,
+		codeOffset: *const u8,
+		codeMetadataOffset: *const u8,
+		length: i32,
+		resultOffset: *const u8,
+		numArguments: i32,
+		argumentsLengthOffset: *const u8,
+		dataOffset: *const u8,
+	) -> i32;
+
+	fn executeOnDestContext(
+		gas: u64,
+		addressOffset: *const u8,
+		valueOffset: *const u8,
+		functionOffset: *const u8,
+		functionLength: i32,
+		numArguments: i32,
+		argumentsLengthOffset: *const u8,
+		dataOffset: *const u8,
+	);
+	fn executeOnDestContextByCaller(
+		gas: u64,
+		addressOffset: *const u8,
+		valueOffset: *const u8,
+		functionOffset: *const u8,
+		functionLength: i32,
+		numArguments: i32,
+		argumentsLengthOffset: *const u8,
+		dataOffset: *const u8,
+	);
+	fn executeOnSameContext(
+		gas: u64,
+		addressOffset: *const u8,
+		valueOffset: *const u8,
+		functionOffset: *const u8,
+		functionLength: i32,
+		numArguments: i32,
+		argumentsLengthOffset: *const u8,
+		dataOffset: *const u8,
+	);
+}
+
+impl SendApi<ArwenBigUint> for ArwenApiImpl {
+	fn direct_egld(&self, to: &Address, amount: &ArwenBigUint, data: &[u8]) {
+		unsafe {
+			let amount_bytes32_ptr = amount.unsafe_buffer_load_be_pad_right(32);
+			let _ = transferValue(
```

Comment:
The return codes of all the C functions here are ignored. Aren't there situations where they may result in errors and should be reported in the rust code (via a panic or transaction error)? Or are they always successful?

Reply:
They will kill the execution anyway.